src/fsharp/ast.fs

```diff
@@ -1272,97 +1293,97 @@ and
-        | SynMemberDefn.AbstractSlot(_,_,m)
-        | SynMemberDefn.Inherit(_,_,m)
-        | SynMemberDefn.ValField(_,m)
-        | SynMemberDefn.AutoProperty(_,_,_,_,_,_,_,_,_,_,m)
-        | SynMemberDefn.NestedType(_,_,m) -> m
+        | SynMemberDefn.Member (range=r)
+        | SynMemberDefn.Interface (range=r)
+        | SynMemberDefn.Open (range=r)
+        | SynMemberDefn.LetBindings (range=r)
+        | SynMemberDefn.ImplicitCtor (range=r)
+        | SynMemberDefn.ImplicitInherit (range=r)
+        | SynMemberDefn.AbstractSlot (range=r)
+        | SynMemberDefn.Inherit (range=r)
+        | SynMemberDefn.ValField (range=r)
+        | SynMemberDefn.AutoProperty (range=r)
+        | SynMemberDefn.NestedType (range=r) -> r
       
 and SynMemberDefns = SynMemberDefn list
 
 and 
     [<NoEquality; NoComparison;RequireQualifiedAccess>]
     SynModuleDecl =
-    | ModuleAbbrev of Ident * LongIdent * range
-    | NestedModule of SynComponentInfo * SynModuleDecls * bool * range
-    | Let of bool * SynBinding list * range
-    | DoExpr of SequencePointInfoForBinding * SynExpr * range 
-    | Types of SynTypeDefn list * range
-    | Exception of SynExceptionDefn * range
-    | Open of LongIdentWithDots * range
-    | Attributes of SynAttributes * range
-    | HashDirective of ParsedHashDirective * range
+    | ModuleAbbrev of id:Ident * longId:LongIdent * range:range
+    | NestedModule of SynComponentInfo * moduleDecls:SynModuleDecls * bool * range:range
+    | Let of bool * bindings:SynBinding list * range:range
+    | DoExpr of spBind:SequencePointInfoForBinding * SynExpr * range:range 
+    | Types of typeDefns:SynTypeDefn list * range:range
+    | Exception of exnDefn:SynExceptionDefn * range:range
+    | Open of dotId:LongIdentWithDots * range:range
+    | Attributes of attributes:SynAttributes * range:range
+    | HashDirective of hashDirectives:ParsedHashDirective * range:range
     | NamespaceFragment of SynModuleOrNamespace 
     member d.Range = 
         match d with 
-        | SynModuleDecl.ModuleAbbrev(_,_,m) 
-        | SynModuleDecl.NestedModule(_,_,_,m)
-        | SynModuleDecl.Let(_,_,m) 
-        | SynModuleDecl.DoExpr(_,_,m) 
-        | SynModuleDecl.Types(_,m)
-        | SynModuleDecl.Exception(_,m)
-        | SynModuleDecl.Open (_,m)
-        | SynModuleDecl.HashDirective (_,m)
-        | SynModuleDecl.NamespaceFragment(SynModuleOrNamespace(_,_,_,_,_,_,m)) 
-        | SynModuleDecl.Attributes(_,m) -> m
+        | SynModuleDecl.ModuleAbbrev (range=r)
+        | SynModuleDecl.NestedModule (range=r)
+        | SynModuleDecl.Let (range=r)
+        | SynModuleDecl.DoExpr (range=r)
+        | SynModuleDecl.Types (range=r)
+        | SynModuleDecl.Exception (range=r)
+        | SynModuleDecl.Open (range=r)
+        | SynModuleDecl.HashDirective (range=r)
+        | SynModuleDecl.NamespaceFragment (SynModuleOrNamespace (range=r)) 
+        | SynModuleDecl.Attributes (range=r) -> r
```

Comment:
Does that mean you want the field names in the type definitions like -
```
| Const of constant:SynConst * m:range
```
 or just to write the bindings in the pattern match cases like
```
| SynModuleDecl.ModuleAbbrev (range=m)
```
If `m` is the standard name does that mean all of the instances of [mWholeExpr](https://github.com/cloudRoutine/FSharp.Compiler.Service/blob/union-fieldnames/src/fsharp/ast.fs#L556), [wholeRange](https://github.com/cloudRoutine/FSharp.Compiler.Service/blob/union-fieldnames/src/fsharp/ast.fs#L562),  [rangeOfWholeExpr](https://github.com/cloudRoutine/FSharp.Compiler.Service/blob/union-fieldnames/src/fsharp/ast.fs#L615), [range](https://github.com/cloudRoutine/FSharp.Compiler.Service/blob/union-fieldnames/src/fsharp/ast.fs#L898), [mWholeAutoProp](https://github.com/cloudRoutine/FSharp.Compiler.Service/blob/union-fieldnames/src/fsharp/ast.fs#L1312), and the few other field names used for `range` should all be changed to `m`?
It seems like vast changes are unavoidable if consistency is the goal. I'm happy to change it to whatever convention you prefer.